balrogscript/src/balrogscript/submitter/cli.py

```diff
@@ -261,6 +306,81 @@ def update_latest():
+            logging.debug("Data sent: %s", None)
+            before = time.time()
+            resp = requests_api.request(method=method, url=url, headers=headers, data=data)
+            try:
+                resp.raise_for_status()
+                return json.loads(resp.content)
+                log.info("Data recieved: %s", json.dumps(existing_release))
+            except requests.HTTPError as excp:
+                logging.error("Caught HTTPError: %s", excp.response.content)
+                raise
+            finally:
+                stats = {"timestamp": time.time(), "method": method.upper(), "url": url, "status_code": resp.status_code, "elapsed_secs": time.time() - before}
+                logging.debug("REQUEST STATS: %s", json.dumps(stats))
+
+        targets = buildbot2updatePlatforms(platform)
+        build_target = targets[0]
+        alias = None
+        if len(targets) > 1:
+            alias = targets[1:]
+            log.debug("alias entry of %s ignored...", json.dumps(alias))
+        data = {"buildID": buildID, "appVersion": appVersion, "platformVersion": extVersion, "displayVersion": appVersion}
+        if isOSUpdate:
+            data["isOSUpdate"] = isOSUpdate
+
+        data.update(self._get_update_data(productName, branch, **updateKwargs))
+
+        if "old-id" in platform:
+            # bug 1366034: support old-id builds
+            # Like 1055305, this is a hack to support two builds with same build target that
+            # require differed't release blobs and rules
+            build_type = "old-id-%s" % self.build_type
+        else:
+            build_type = self.build_type
+
+        # wrap operations into "atomic" functions that can be retried
+        def update_data(url, existing_release, existing_locale_data):
+            # If the  partials are already a subset of the blob and the
+            # complete MAR is the same, skip the submission
+            skip_submission = bool(
+                existing_locale_data
+                and existing_locale_data.get("completes") == data.get("completes")
+                and all(p in existing_locale_data.get("partials", []) for p in data.get("partials", []))
+            )
+            if skip_submission:
+                log.warning("Dated data didn't change, skipping update")
+                return
+            # explicitly pass data version
+            blob = {"blob": {"platforms": {build_target: {"locales": {locale: data}}}}, "old_data_versions": {"platforms": {build_target: {"locales": {}}}}}
+            if existing_release.get("old_data_versions", {}).get("platforms", {}).get(build_target, {}).get("locales", {}).get(locale):
+                blob["old_data_versions"]["platforms"][build_target]["locales"][locale] = existing_release["old_data_versions"]["platforms"][build_target][
+                    "locales"
+                ][locale]
+            do_request(url=url, method="post", data=json.dumps(blob))
```

Comment:
You can pass this as `json` instead of dumping to `data`. This will also ensure proper headers are set.